src/btree/bt_handle.c

```diff
@@ -133,16 +133,13 @@ __wt_btree_close(WT_SESSION_IMPL *session)
 {
 	WT_BM *bm;
 	WT_BTREE *btree;
-	WT_DATA_HANDLE *dhandle;
 	WT_DECL_RET;
 
-	dhandle = session->dhandle;
 	btree = S2BT(session);
 
 	if ((bm = btree->bm) != NULL) {
 		/* Unload the checkpoint, unless it's a special command. */
-		if (F_ISSET(dhandle, WT_DHANDLE_OPEN) &&
-		    !F_ISSET(btree,
+		if (!F_ISSET(btree,
```

Comment:
This is the real fix: unload the checkpoint if we close a tree before `WT_DHANDLE_OPEN` is set.